File: Criando Bacterias/CriandoBacterias.py
objetivo = int(input())

valores = {1:1, 2:1, 3:2, 4:1, 5:2, 6:2, 7:3, 8:1, 9:24,}
count =0

while objetivo>9:
    if objetivo%2!=0:
        count+=1
        objetivo-=1
    objetivo=objetivo/2

print(valores[objetivo]+count)

File: Criando Bacterias/CriandoBacterias.c
#include <stdio.h>
int main()
{
    int objetivo;
    int count=0;
    scanf("%d", &objetivo);

    while (objetivo>9){
    if (objetivo%2!=0)
    {
    count+=1;
    objetivo-=1;
    }
    objetivo=objetivo/2;
    }

    if(objetivo==1 || objetivo ==2 || objetivo ==4 || objetivo ==8)
    {
        printf("%d", 1+count);
    }
    else if(objetivo==7)
    {
        printf("%d", 3+count);
    }
    else
    {
        printf("%d", 2+count);
    }
    return 0;
}

